### PR TITLE
stop using deprecated Doctrine persistence classes

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/ManagerRegistry.php [hunk 1]
```diff
@@ -12,17 +12,30 @@
...
+    }
+}
+
 /**
  * References Doctrine connections and entity/document managers.
  *
  * @author  Lukas Kahwe Smith <smith@pooteeweet.org>
  */
-abstract class ManagerRegistry extends LegacyAbstractManagerRegistry implements ContainerAwareInterface
+trait ManagerRegistryTrait
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/ManagerRegistry.php]: let's mark the trait as internal

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Comment by gmponos** [comment]
I still get deprecations

image

```
> composer show | grep symfony -i

symfony/asset                                               v5.0.2             Symfony Asset Component
symfony/browser-kit                                         v5.0.2             Symfony BrowserKit Component
symfony/cache                                               v5.0.2             Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts                                     v2.0.1             Generic abstractions related to caching
symfony/config                                              v5.0.2             Symfony Config Component
symfony/console                                             v5.0.2             Symfony Console Component
symfony/css-selector                                        v5.0.2             Symfony CssSelector Component
symfony/debug-bundle                                        v5.0.2             Symfony DebugBundle
symfony/dependency-injection                                v5.0.2             Symfony DependencyInjection Component
symfony/doctrine-bridge                                     v5.0.2             Symfony Doctrine Bridge
symfony/dom-crawler                                         v5.0.2             Symfony DomCrawler Component
symfony/dotenv                                              v5.0.2             Registers environment variables from a .env file
symfony/error-handler                                       v5.0.2             Symfony ErrorHandler Component
symfony/event-dispatcher                                    v5.0.2             Symfony EventDispatcher Component
symfony/event-dispatcher-contracts                          v2.0.1             Generic abstractions related to dispatching event
symfony/expression-language                                 v5.0.2             Symfony ExpressionLanguage Component
symfony/filesystem                                          v5.0.2             Symfony Filesystem Component
symfony/finder                                              v5.0.2             Symfony Finder Component
symfony/flex                                                v1.6.0             Composer plugin for Symfony
symfony/form                                                v5.0.2             Symfony Form Component
symfony/framework-bundle                                    v5.0.2             Symfony FrameworkBundle
symfony/http-foundation                                     v5.0.2             Symfony HttpFoundation Component
symfony/http-kernel                                         v5.0.2             Symfony HttpKernel Component
symfony/inflector                                           v5.0.2             Symfony Inflector Component
symfony/intl                                                v5.0.2             A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/maker-bundle                                        v1.14.3            Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about writing...
symfony/messenger                                           v5.0.2             Symfony Messenger Component
symfony/mime                                                v5.0.2             A library to manipulate MIME messages
symfony/monolog-bridge                                      v5.0.2             Symfony Monolog Bridge
symfony/monolog-bundle                                      v3.5.0             Symfony MonologBundle
symfony/options-resolver                                    v5.0.2             Symfony OptionsResolver Component
symfony/polyfill-ctype                                      v1.13.1            Symfony polyfill for ctype functions
symfony/polyfill-intl-icu                                   v1.13.1            Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn                                   v1.13.1            Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-mbstring                                   v1.13.1            Symfony polyfill for the Mbstring extension
symfony/polyfill-php72                                      v1.13.1            Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73                                      v1.13.1            Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/process                                             v5.0.2             Symfony Process Component
symfony/property-access                                     v5.0.2             Symfony PropertyAccess Component
symfony/proxy-manager-bridge                                v5.0.2             Symfony ProxyManager Bridge
symfony/routing                                             v5.0.2             Symfony Routing Component
symfony/security-bundle                                     v5.0.2             Symfony SecurityBundle
symfony/security-core                                       v5.0.2             Symfony Security Component - Core Library
symfony/security-csrf                                       v5.0.2             Symfony Security Component - CSRF Library
symfony/security-guard                                      v5.0.2             Symfony Security Component - Guard
symfony/security-http                                       v5.0.2             Symfony Security Component - HTTP Integration
symfony/serializer                                          v5.0.2             Symfony Serializer Component
symfony/service-contracts                                   v2.0.1             Generic abstractions related to writing services
symfony/stopwatch                                           v5.0.2             Symfony Stopwatch Component
symfony/swiftmailer-bundle                                  v3.4.0             Symfony SwiftmailerBundle
symfony/translation                                         v5.0.2             Symfony Translation Component
symfony/translation-contracts                               v2.0.1             Generic abstractions related to translation
symfony/twig-bridge                                         v5.0.2             Symfony Twig Bridge
symfony/twig-bundle                                         v5.0.2             Symfony TwigBundle
symfony/validator                                           v5.0.2             Symfony Validator Component
symfony/var-dumper                                          v5.0.2             Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter                                        v5.0.2             A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-profiler-bundle                                 v5.0.2             Symfony WebProfilerBundle
symfony/webpack-encore-bundle                               v1.7.2             Integration with your Symfony app & Webpack Encore!
symfony/yaml                                                v5.0.2             Symfony Yaml Component
```

Are the deprecations fixed here only some of them?

**Comment by nicolas-grekas** [comment]
@gmponos please have a look at the stacj trace and report them to the project that triggers them. That's not Symfony anymoe.

**Comment by stof** [comment]
these are not from Symfony, but from the Doctrine ORM (where this work is in progress)

**Comment by gmponos** [comment]
thank you sorry for the noise..

**Comment by protracks** [comment]
I get these depreciations right after a composer update, but when i do php bin/console cache:clear they go away?

**Comment by nicolas-grekas** [comment]
I'm locking this PR because the issue is fixed and deprecations from doctrine/orm or any other package are not ours to handle. Please ask on the appropriate repositories.

**Pull request merged**